### PR TITLE
Disable FP8 in Mcore integration test on older GPUs

### PR DESCRIPTION
**Pull request by timmoon10**
# Description

We have experienced test failures in the Mcore integration tests on older GPUs, so this PR checks the GPU arch before enabling FP8. It also generates synthetic data so it can be run without needing to access a specific data path.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

- Disable FP8 in Mcore integration test on older GPUs
- Remove data dependency in Mcore integration test

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: qa/L1_pytorch_mcore_integration/test.sh [hunk 1]
```diff
@@ -8,13 +8,27 @@ set -e
...
+# Check whether FP8 is supported
+DEVICE_ARCH=$(nvidia-smi --query-gpu=compute_cap --format=csv,noheader | head -n 1 | sed 's/[^0-9]//g')
+if [[ ${DEVICE_ARCH} -ge 89 ]]; then
+    WITH_FP8=1
+fi
+
 # Download Megatron-LM if needed
 if [ ! -d "${MCORE_PATH}" ]; then
     pushd $(dirname ${MCORE_PATH})
     git clone -b core_r0.9.0 https://github.com/NVIDIA/Megatron-LM.git Megatron-LM
```
Comment (sudhakarsingh27) [inline qa/L1_pytorch_mcore_integration/test.sh:20]: I guess then we have to manually update this branch. Is this okay?

File: qa/L1_pytorch_mcore_integration/test.sh [hunk 2]
```diff
@@ -8,13 +8,27 @@ set -e
...
     popd
 fi
 
+# Create mock vocab
+VOCAB_FILE=${TE_PATH}/qa/L1_pytorch_mcore_integration/vocab.json
+printf "" > ${VOCAB_FILE}
+printf "{" >> ${VOCAB_FILE}
+printf "\"<|endoftext|>\": 0" >> ${VOCAB_FILE}
+seq 1 4095 | awk '{ printf(", \"%d\": %d", $1, $1) }' >> ${VOCAB_FILE}
+printf "}" >> ${VOCAB_FILE}
```
Comment (sudhakarsingh27) [inline qa/L1_pytorch_mcore_integration/test.sh:30]: Wow, that's enough to generate a vocab file!

File: qa/L1_pytorch_mcore_integration/test.sh [hunk 3]
```diff
@@ -40,17 +54,17 @@ ${MCORE_PATH}/pretrain_gpt.py
...
 --micro-batch-size 1
 --global-batch-size 8
 --train-iters 10
 --eval-iters 10
 --lr 1e-4
 --mock-data
---vocab-file /data/gpt3/pile-cc1-cc2-shuf/bpe/gpt2-vocab.json
---merge-file /data/gpt3/pile-cc1-cc2-shuf/bpe/gpt2-merges.txt
+--vocab-file ${VOCAB_FILE}
+--merge-file ${TE_PATH}/qa/L1_pytorch_mcore_integration/merges.txt
```
Comment (sudhakarsingh27) [inline qa/L1_pytorch_mcore_integration/test.sh:65]: Not sure what's the purpose of this file, but is it okay for it to only have version info?

**Review by sudhakarsingh27 — APPROVED** [review]
Only minor nitpicks/questions. Ow, lgtm

File: qa/L1_pytorch_mcore_integration/test.sh (hunk 1 above)
Comment (timmoon10) [reply qa/L1_pytorch_mcore_integration/test.sh:20]: I think there's an argument for just downloading the Megatron-LM `main` branch, but that is orthogonal to this PR.

File: qa/L1_pytorch_mcore_integration/test.sh (hunk 2 above)
Comment (timmoon10) [reply qa/L1_pytorch_mcore_integration/test.sh:30]: This is somewhat specific to the mock GPT dataset: https://github.com/NVIDIA/Megatron-LM/blob/bd677bfb13ac2f19deaa927adc6da6f9201d66aa/megatron/core/datasets/gpt_dataset.py#L693
Each data sample is just generated with `arange`, with a max sequence length of 4096.

File: qa/L1_pytorch_mcore_integration/test.sh (hunk 3 above)
Comment (timmoon10) [reply qa/L1_pytorch_mcore_integration/test.sh:65]: I'm treating `merges.txt` as a magic config for the tokenizer: https://github.com/huggingface/transformers/issues/1083#issuecomment-524303077

**Pull request merged**